apps/console/src/features/groups/components/edit-group/edit-group-roles-updated.tsx

```diff
@@ -0,0 +1,151 @@
+import { IdentifiableComponentInterface, RolesMemberInterface } from "@wso2is/core/models";
+import { EmphasizedSegment, EmptyPlaceholder, Heading } from "@wso2is/react-components";
+import React, {
+    FunctionComponent,
+    HTMLAttributes,
+    ReactElement,
+    useEffect,
+    useState
+} from "react";
+import { useTranslation } from "react-i18next";
+import { getEmptyPlaceholderIllustrations } from "../../../core/configs/ui";
+import { GroupsInterface } from "../../models/groups";
+import { AutoCompleteRenderOption } from "../group-common-components/auto-complete-render-option";
+import { RenderChip } from "../group-common-components/render-chip";
+
+interface EditGroupRolesPropsInterface extends IdentifiableComponentInterface {
+    /**
+     * User profile
+     */
+    group: GroupsInterface;
+}
+
+export const EditGroupRoles: FunctionComponent<EditGroupRolesPropsInterface> = (
+    props: EditGroupRolesPropsInterface
+): ReactElement => {
+
+    const {
+        group
+    } = props;
+
+    const { t } = useTranslation();
+
+    const [ initialSelectedRolesOptions, setInitialSelectedRolesOptions ] = useState<RolesMemberInterface[]>(undefined);
+    const [ activeOption, setActiveOption ] = useState<RolesMemberInterface>(undefined);
+    const [ showEmptyRolesListPlaceholder, setShowEmptyRolesListPlaceholder ] = useState<boolean>(false);
+
+    /**
+     * Set initial selected roles options
+     */
+    useEffect(() => {
+        if ( group?.roles?.length > 0 ) {
+            setInitialSelectedRolesOptions(group.roles.map((role: RolesMemberInterface) => role));
+        } else {
+            setInitialSelectedRolesOptions([]);
+            setShowEmptyRolesListPlaceholder(true);
+        }
+    }, [ group ]);
+
+    /**
+     * Get the place holder components.
+     * 
+     * @returns - place holder components
+     */
+    const getPlaceholders = () => {
+        if (showEmptyRolesListPlaceholder) {
+            return (
+                // TODO: Need to replace the i18N with the correct one.
+                <EmptyPlaceholder
+                    subtitle={ 
+                        [ t("console:manage.features.roles.edit.groups.placeholders.emptyPlaceholder.subtitles.0") ]
+                    }
+                    title={ t("console:manage.features.roles.edit.groups.placeholders.emptyPlaceholder.title") }
+                    image={ getEmptyPlaceholderIllustrations().emptyList }
+                    imageSize="tiny"
+                />
+            );
+        }
+    };
+
+    return (
+        <EmphasizedSegment padded="very">
+            <Heading as="h4">
+                { t("console:manage.features.user.updateUser.roles.editRoles.heading") }
+                <Heading subHeading ellipsis as="h6">
+                    { /* TODO: Need to replace this with "View assigned roles for the group." */ }
+                    { t("console:manage.features.groups.edit.roles.subHeading") }
```

Comment:
Shall we update these in this PR itself :) ?

Reply:
If these are updated, it will affect the current UI, we cannot affect the current UI until the fix is fully implemented.